src/test/get_mediapipe_graph_metadata_response_test.cpp

```diff
@@ -107,3 +122,179 @@ TEST_F(GetMediapipeGraphMetadataResponse, BasicResponseMetadata) {
+            return ovms::MediapipeGraphDefinition::status;
+        }
+    };
+
+    MockMediapipeGraphDefinitionGetInputsOutputsInfo graphDefinition;
+    KFSModelMetadataResponse response;
+    ConstructorEnabledModelManager manager;
+};
+
+TEST_F(MediapipeGraphDefinitionMetadataResponseBuild, GraphNotLoadedAnymore) {
+    graphDefinition.getGraphDefinitionStatus().handle(ovms::RetireEvent());
+    auto status = ovms::KFSInferenceServiceImpl::buildResponse(graphDefinition, &response);
+    ASSERT_EQ(status, ovms::StatusCode::MEDIAPIPE_DEFINITION_NOT_LOADED_ANYMORE) << status.string();
+}
+
+TEST_F(MediapipeGraphDefinitionMetadataResponseBuild, GraphNotLoadedYet) {
+    graphDefinition.getGraphDefinitionStatus().handle(ovms::UsedModelChangedEvent());
+    graphDefinition.getGraphDefinitionStatus().handle(ovms::ValidationFailedEvent());
+    auto status = ovms::KFSInferenceServiceImpl::buildResponse(graphDefinition, &response);
+    ASSERT_EQ(status, ovms::StatusCode::MEDIAPIPE_DEFINITION_NOT_LOADED_YET) << status.string();
+    graphDefinition.getGraphDefinitionStatus().handle(ovms::UsedModelChangedEvent());
+    ASSERT_EQ(ovms::KFSInferenceServiceImpl::buildResponse(graphDefinition, &response), ovms::StatusCode::MEDIAPIPE_DEFINITION_NOT_LOADED_YET);
+}
+
+TEST_F(MediapipeGraphDefinitionMetadataResponseBuild, GraphAvailableOrAvailableRequiringRevalidation) {
+    graphDefinition.getGraphDefinitionStatus().handle(ovms::UsedModelChangedEvent());
+    EXPECT_EQ(ovms::KFSInferenceServiceImpl::buildResponse(graphDefinition, &response), ovms::StatusCode::OK);
+}
+
+const char* dummy_config = R"({
+      "model_config_list":[
+        {
+          "config":{
+            "name":"dummy",
+            "base_path": "/tmp/test_cl_models/model1"
+          }
+        }
+      ]
+    })";
+
+class TestImplGetModelStatus : public ::testing::Test {
+public:
+    ConstructorEnabledModelManager manager;
+    std::string cl_models_path;
+    std::string cl_model_1_path;
+    std::string cl_model_2_path;
+
+    void SetUpSingleModel(std::string modelPath, std::string modelName) {
+        char* n_argv[] = {(char*)"ovms", (char*)"--model_path", (char*)modelPath.data(), (char*)"--model_name", (char*)modelName.data(), (char*)"--file_system_poll_wait_seconds", (char*)"0"};
+        int arg_count = 7;
+        ovms::Config::instance().parse(arg_count, n_argv);
+    }
+
+    void SetUp() {
+        const ::testing::TestInfo* const test_info =
+            ::testing::UnitTest::GetInstance()->current_test_info();
+
+        cl_models_path = "/tmp/" + std::string(test_info->name());
+        cl_model_1_path = cl_models_path + "/model1/";
+        cl_model_2_path = cl_models_path + "/model2/";
+
+        const std::string FIRST_MODEL_NAME = "dummy";
+        const std::string SECOND_MODEL_NAME = "dummy_new";
+
+        std::filesystem::remove_all(cl_models_path);
+        std::filesystem::create_directories(cl_model_1_path);
```

Comment:
Either use TestWithTempDir or ensure cleanup of all tmp directories

Reply:
Cleanup ensured with test TearDown cleanup